L1Trigger/TrackerDTC/plugins/Analyzer.cc

```diff
@@ -0,0 +1,508 @@
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "Geometry/CommonTopologies/interface/PixelGeomDetUnit.h"
+#include "Geometry/CommonTopologies/interface/PixelTopology.h"
+#include "SimDataFormats/TrackingAnalysis/interface/TrackingParticle.h"
+#include "SimTracker/TrackTriggerAssociation/interface/TTClusterAssociationMap.h"
+#include "SimTracker/Common/interface/TrackingParticleSelector.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+#include "DataFormats/Common/interface/Ptr.h"
+#include "DataFormats/L1TrackTrigger/interface/TTTypes.h"
+#include "DataFormats/L1TrackTrigger/interface/TTDTC.h"
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+#include "DataFormats/GeometrySurface/interface/Plane.h"
+#include "DataFormats/SiStripDetId/interface/StripSubdetector.h"
+
+#include "L1Trigger/TrackerDTC/interface/Settings.h"
+#include "L1Trigger/TrackerDTC/interface/TTDTCConverter.h"
+
+#include <TProfile.h>
+#include <TProfile2D.h>
+#include <TH1F.h>
+#include <TH2F.h>
+#include <TEfficiency.h>
+
+#include <vector>
+#include <map>
+#include <utility>
+#include <set>
+#include <algorithm>
+#include <cmath>
+#include <numeric>
+#include <array>
+#include <initializer_list>
+#include <sstream>
+
+using namespace std;
+using namespace edm;
+
+namespace trackerDTC {
+
+  // mc truth types
+  typedef TTClusterAssociationMap<Ref_Phase2TrackerDigi_> TTClusterAssMap;
+  typedef edm::Ptr<TrackingParticle> TPPtr;
+  // stub resolution plots helper
+  enum Resolution { R, Phi, Z, NumResolution };
+  constexpr initializer_list<Resolution> AllResolution = {R, Phi, Z};
+  constexpr auto NameResolution = {"R", "Phi", "Z"};
+  inline string name(Resolution r) { return string(*(NameResolution.begin() + r)); }
+  // max tracking efficiency plots helper
+  enum Efficiency { Phi0, Pt, InvPt, D0, Z0, Eta, NumEfficiency };
+  constexpr initializer_list<Efficiency> AllEfficiency = {Phi0, Pt, InvPt, D0, Z0, Eta};
+  constexpr auto NameEfficiency = {"Phi0", "Pt", "InvPt", "D0", "Z0", "Eta"};
+  inline string name(Efficiency e) { return string(*(NameEfficiency.begin() + e)); }
+
+  /*! \class  trackerDTC::Analyzer
+   *  \brief  Class to analyze hardware like structured TTStub Collection used by Track Trigger emulators
+   *  \author Thomas Schuh
+   *  \date   2020, Apr
+   */
+  class Analyzer : public one::EDAnalyzer<one::WatchRuns, one::SharedResources> {
```

Comment:
this seems like it should migrate toward the validation subsystem eventually (as always, sooner is better than later)

Reply:
I will get in contact with the person making DQM plots in our group to asses if, what and how. Will come back when we have a plan but I cannot promise that we will do that quick enough for 11.1
Is it ok if that lives for the short term here?

Reply:
Yes, we've allowed it before (on a temporary basis)